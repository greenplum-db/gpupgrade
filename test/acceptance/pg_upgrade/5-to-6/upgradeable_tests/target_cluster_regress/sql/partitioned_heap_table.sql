-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

--------------------------------------------------------------------------------
-- Validate that the upgradeable objects are functional post-upgrade
--------------------------------------------------------------------------------

SELECT * FROM p_basic;

SELECT * FROM p_add_partition_test;

SELECT * FROM p_add_list_partition_test;

SELECT * FROM p_split_partition_test;
INSERT INTO p_split_partition_test SELECT i, i FROM generate_series(6,10)i;
ALTER TABLE p_split_partition_test SPLIT DEFAULT PARTITION START(6) END(10) INTO (PARTITION second_split, PARTITION extra);
SELECT * FROM p_split_partition_test;
SELECT parname, parisdefault FROM pg_partition_rule pr JOIN pg_partition p ON pr.paroid = p.oid WHERE p.parrelid = 'p_split_partition_test'::regclass AND pr.parname != '';

SELECT id, age FROM p_subpart_heap_1_prt_partition_id_2_prt_subpartition_age_first;
SELECT id, age FROM p_subpart_heap_1_prt_partition_id_2_prt_subpartition_age_second;
SELECT id, age FROM p_subpart_heap;

SELECT b, c FROM dropped_column WHERE a=10;

SELECT b, c FROM root_has_dropped_column WHERE a=10;

SELECT c, d FROM dropped_and_added_column WHERE a=10;

SELECT c.relname, pg_catalog.pg_get_userbyid(c.relowner) as owner
FROM pg_class c
WHERE relname like 'p_alter_owner%';
