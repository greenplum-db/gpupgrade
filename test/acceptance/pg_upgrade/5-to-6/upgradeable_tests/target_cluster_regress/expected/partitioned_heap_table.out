-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

--------------------------------------------------------------------------------
-- Validate that the upgradeable objects are functional post-upgrade
--------------------------------------------------------------------------------

SELECT * FROM p_basic;
 id | name 
----+------
 1  | Jane 
 2  | John 
(2 rows)

SELECT * FROM p_add_partition_test;
 a | b 
---+---
 1 | 1 
 2 | 1 
 1 | 2 
 1 | 3 
(4 rows)

SELECT * FROM p_add_list_partition_test;
 a | b 
---+---
 1 | 1 
 2 | 1 
 1 | 2 
 1 | 3 
(4 rows)

SELECT * FROM p_split_partition_test;
 a | b 
---+---
 1 | 1 
 2 | 2 
 3 | 3 
 4 | 4 
 5 | 5 
(5 rows)
INSERT INTO p_split_partition_test SELECT i, i FROM generate_series(6,10)i;
INSERT 5
ALTER TABLE p_split_partition_test SPLIT DEFAULT PARTITION START(6) END(10) INTO (PARTITION second_split, PARTITION extra);
ALTER
SELECT * FROM p_split_partition_test;
 a  | b  
----+----
 1  | 1  
 10 | 10 
 2  | 2  
 3  | 3  
 4  | 4  
 5  | 5  
 6  | 6  
 7  | 7  
 8  | 8  
 9  | 9  
(10 rows)
SELECT parname, parisdefault FROM pg_partition_rule pr JOIN pg_partition p ON pr.paroid = p.oid WHERE p.parrelid = 'p_split_partition_test'::regclass AND pr.parname != '';
 parname      | parisdefault 
--------------+--------------
 extra        | t            
 second_split | f            
 splitted     | f            
(3 rows)

SELECT id, age FROM p_subpart_heap_1_prt_partition_id_2_prt_subpartition_age_first;
 id | age 
----+-----
 1  | 10  
(1 row)
SELECT id, age FROM p_subpart_heap_1_prt_partition_id_2_prt_subpartition_age_second;
 id | age 
----+-----
 2  | 20  
(1 row)
SELECT id, age FROM p_subpart_heap;
 id | age 
----+-----
 1  | 10  
 2  | 20  
(2 rows)

SELECT b, c FROM dropped_column WHERE a=10;
 b  | c  
----+----
 10 | 10 
 10 | 10 
(2 rows)

SELECT b, c FROM root_has_dropped_column WHERE a=10;
 b  | c  
----+----
 10 | 10 
(1 row)

SELECT c, d FROM dropped_and_added_column WHERE a=10;
 c  | d  
----+----
 10 | 10 
 10 | 10 
(2 rows)

SELECT c.relname, pg_catalog.pg_get_userbyid(c.relowner) as owner FROM pg_class c WHERE relname like 'p_alter_owner%';
 relname               | owner                   
-----------------------+-------------------------
 p_alter_owner         | upgradable_objects_role 
 p_alter_owner_1_prt_1 | upgradable_objects_role 
 p_alter_owner_1_prt_2 | upgradable_objects_role 
(3 rows)

INSERT INTO equal_operator_not_in_search_path_table VALUES (1, '(1,1)');
INSERT 1
SELECT * FROM equal_operator_not_in_search_path_table;
 a | b     
---+-------
 1 | (1,1) 
(1 row)

INSERT INTO equal_operator_not_in_search_path_table_multi_key VALUES (1, '(1,1)', 1);
INSERT 1
SELECT * FROM equal_operator_not_in_search_path_table_multi_key;
 a | b     | c 
---+-------+---
 1 | (1,1) | 1 
(1 row)
